property/index.js

```diff
@@ -166,3 +156,68 @@ module.exports = yeoman.generators.Base.extend({
 
   saveProject: actions.saveProject
 });
+
+function coerceDefault(def, value) {
+  var isSupported = typeChoices.indexOf(def.type) !== -1;
+  debug('property type "%s" >> supported: %s', def.type, isSupported);
+
+  if (typeof value === 'string' && isSupported) {
+    switch ((def.type || '').toLowerCase()) {
+      case 'string':
+        if (value === 'uuid' || value === 'guid'){
+          def.defaultFn = value;
+        } else {
+          def.default = value;
+        }
+        break;
+      case 'number':
+        def.default = Number(value);
+        break;
+      case 'boolean':
+        if (['true', '1', 't'].indexOf(value) !== -1 ){
+          def.default = true;
+        } else {
+          def.default = false;
+        }
+        break;
+      case 'object':
+        def.default = JSON.parse(value);
+        break;
+      case 'array':
+        def.default = value.replace(/[\s,]+/g, ',').split(',');
```

Comment:
This will not work very well for array of strings. I am proposing to support two formats: JSON-encoded array, e.g. `[1,2]` or `["a,b", "c"]`; and your current comma-separated list, e.g. `1,2`.